test/Sema/superfluously-public-imports.swift

```diff
@@ -96,6 +97,9 @@ public struct NonPublicExtendedType {}
 //--- UnusedImport.swift
 
 //--- UnusedPackageImport.swift
+//--- ExtendedPackageTypeImport.swift
+
+public struct ExtendedPackageType {}
```

Comment:
Re: comment above, `package struct P {}` could be extended without any members and the warning will still be shown.

Reply:
I believe that warning would be appropriate then as an empty extension would be internal or below.
```
package import DefinesP // warning: package import of 'DefinesP' was not used in package declarations

extension P {} 
```

It would fall in the same scenario as extending the type with internal decls where there's no requirement to make the import package or public as these services are not visible to other modules. Does that make sense or am I missing something?

Reply:
I see. For some reason I thought the default access level of the extension is whatever access level its nominal type has, but it looks to be `internal` when not explicitly marked. FYI package exportability PR https://github.com/apple/swift/pull/73161

Reply:
I expect #73161 may be more challenging to land as it adds errors. So I'll merge this one but please feel free to revert it as part of #73161 if you see it interfering with the diagnostics.